R/to_integer.R

```diff
@@ -274,8 +272,17 @@ to_integer.gs_design <- function(x, sample_size = TRUE, round_up_final = TRUE, .
     return(x)
   }
 
+  if (ratio < 0) {
+    stop("The ratio must be non-negative.")
+  }
+
+  if (!is_wholenumber(ratio)) {
+    message("The output sample size is not a multiple of (ratio + 1).")
```

Comment:
Should the message mention that `ratio` is being changed to zero? It's not obvious to me what action is being taken.

```suggestion
    message("The output sample size is not a multiple of (ratio + 1). Setting ratio to zero.")
```